File: src/BlazorWith3d/BlazorWith3d.Shared/IBinaryApi.cs
using System;
using System.Threading.Tasks;

namespace BlazorWith3d.Shared
{
    public interface IBinaryApi 
    {
        Func<ArraySegment<byte>,ValueTask>? MainMessageHandler { get; set; }
        ValueTask SendMessage(IBufferWriterWithArraySegment<byte> bytes);
        Func<ArraySegment<byte>,ValueTask<IBufferWriterWithArraySegment<byte>>>? MainMessageWithResponseHandler { get; set; }
        ValueTask<ArraySegment<byte>> SendMessageWithResponse(IBufferWriterWithArraySegment<byte> bytes);
    }
}

File: samples/BlazorWith3d.ExampleApp.Client.Shared/scripts/IBinaryApi.ts

export interface IBinaryApi {

    mainMessageHandler?: (bytes: Uint8Array) => Promise<void>;

    mainMessageWithResponseHandler?: (bytes: Uint8Array) => Promise<Uint8Array>;

    sendMessage(bytes: Uint8Array): Promise<void>;
    sendMessageWithResponse(bytes: Uint8Array): Promise<Uint8Array>;
}

